src/openfunction/runtime.ts

```diff
@@ -243,4 +261,216 @@ class DaprRuntime extends OpenFunctionRuntime {
 
     return Promise.allSettled(actions);
   }
+
+  /**
+   * Save the data to the state store
+   * @param data The data for save operation
+   * @param db The state store to save the data
+   * @returns The promise of the save action being executed.
+   */
+  #save(data: object, db?: string): Promise<object> {
+    if (!this.context.states) {
+      return Promise.resolve([]);
+    }
+    const states = this.context.states;
+    const actions = Object.keys(states)
+      .filter(v => !db || v === db)
+      .map((storeName: string) => {
+        const component: ComponentSpec = states[storeName];
+        if (ContextUtils.IsStateComponent(component)) {
```

Comment:
[suggestion] This type checking could be merged with the above-mentioned store picking, and it's better to use a quick exit `if () return ...` instead of multiple hierarchies.